### PR TITLE
Make ODataParameters members visible and rename ODataParams to ODataParameters

### PR DESCRIPTION
**Pull request by LucGenetier**

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Connectors/ODataQueryableTableValue.cs [hunk 1]
```diff
@@ -47,68 +47,70 @@ internal override TableValue FirstN(int n)
...
 
-    public readonly struct ODataParams
+    public readonly struct ODataParameters
     {
         // Missing parameters: skip, skipToken, expand, search, select, apply
-        private readonly bool _count;
-        private readonly string _filter;
-        private readonly string _orderby;
-        private readonly int _top;
+        public readonly bool Count;
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/ODataQueryableTableValue.cs:53]: >public readonly bool Count; [](http://example.com/codeflow?start=8&length=27)

nit - fields shouldn't be public . Properties instead. Like:

public bool Count {get; private set; }

**Comment by BruceHaley** [comment]
:x: 5 Binary Compatibility issues for **Microsoft.PowerFx.Connectors** compared against [version 0.2.2-preview.20220831-319648](https://www.nuget.org/packages/Microsoft.PowerFx.Connectors/0.2.2-preview.20220831-319648).
<details>

```
TypesMustExist : Type 'Microsoft.PowerFx.Connectors.ODataParams' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'Microsoft.PowerFx.Connectors.ODataParams Microsoft.PowerFx.Connectors.ODataQueryableTableValue.ODataParams' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'Microsoft.PowerFx.Connectors.ODataQueryableTableValue..ctor(Microsoft.PowerFx.Types.TableType, Microsoft.PowerFx.Connectors.ODataParams)' does not exist in the implementation but it does exist in the contract.
CannotAddAbstractMembers : Member 'Microsoft.PowerFx.Connectors.ODataQueryableTableValue.WithParameters(Microsoft.PowerFx.Connectors.ODataParameters)' is abstract in the implementation but is missing in the contract.
MembersMustExist : Member 'Microsoft.PowerFx.Connectors.ODataQueryableTableValue.WithParameters(Microsoft.PowerFx.Connectors.ODataParams)' does not exist in the implementation but it does exist in the contract.
```

</details>
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by BruceHaley** [comment]
:x: 5 Binary Compatibility issues for **Microsoft.PowerFx.Connectors** compared against [version 0.2.2-preview.20220901-319852](https://www.nuget.org/packages/Microsoft.PowerFx.Connectors/0.2.2-preview.20220901-319852).
<details>

```
TypesMustExist : Type 'Microsoft.PowerFx.Connectors.ODataParams' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'Microsoft.PowerFx.Connectors.ODataParams Microsoft.PowerFx.Connectors.ODataQueryableTableValue.ODataParams' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'Microsoft.PowerFx.Connectors.ODataQueryableTableValue..ctor(Microsoft.PowerFx.Types.TableType, Microsoft.PowerFx.Connectors.ODataParams)' does not exist in the implementation but it does exist in the contract.
CannotAddAbstractMembers : Member 'Microsoft.PowerFx.Connectors.ODataQueryableTableValue.WithParameters(Microsoft.PowerFx.Connectors.ODataParameters)' is abstract in the implementation but is missing in the contract.
MembersMustExist : Member 'Microsoft.PowerFx.Connectors.ODataQueryableTableValue.WithParameters(Microsoft.PowerFx.Connectors.ODataParams)' does not exist in the implementation but it does exist in the contract.
```

</details>
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by BruceHaley** [comment]
:x: 5 Binary Compatibility issues for **Microsoft.PowerFx.Connectors** compared against [version 0.2.2-preview.20220901-319852](https://www.nuget.org/packages/Microsoft.PowerFx.Connectors/0.2.2-preview.20220901-319852).
<details>

```
TypesMustExist : Type 'Microsoft.PowerFx.Connectors.ODataParams' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'Microsoft.PowerFx.Connectors.ODataParams Microsoft.PowerFx.Connectors.ODataQueryableTableValue.ODataParams' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'Microsoft.PowerFx.Connectors.ODataQueryableTableValue..ctor(Microsoft.PowerFx.Types.TableType, Microsoft.PowerFx.Connectors.ODataParams)' does not exist in the implementation but it does exist in the contract.
CannotAddAbstractMembers : Member 'Microsoft.PowerFx.Connectors.ODataQueryableTableValue.WithParameters(Microsoft.PowerFx.Connectors.ODataParameters)' is abstract in the implementation but is missing in the contract.
MembersMustExist : Member 'Microsoft.PowerFx.Connectors.ODataQueryableTableValue.WithParameters(Microsoft.PowerFx.Connectors.ODataParams)' does not exist in the implementation but it does exist in the contract.
```

</details>
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Pull request merged**